### PR TITLE
Add attendance mode to form

### LINKED ISSUE #332: Add option for offsite and remote attendance
Event creators should be able to select whether they want their event attendance mode to be either onsite or remote or both at the point of creating the event.

### PR DESCRIPTION
**Pull request by trymebytes**
Fixes #332 
In this PR we had another field to the event form that allows users choose the attendance mode for their event.

Screenshot 2024-07-10 at 15 14 38

### CODE REVIEW

**Comment by akirk** [comment]
I think we need to find a way to discourage remote attendees for onsite-only events. Maybe like this?

Screenshot 2024-07-11 at 11 10 12

**Comment by akirk** [comment]
Regarding spelling, I am not sure if we need to spell this `On Site` or `on-site` but I don't think it can be `Onsite`.

**Comment by trymebytes** [comment]
I would go with `On-site`

File: templates/parts/event-form.php [hunk 1]
```diff
@@ -71,6 +71,17 @@
 				?>
 			</select>
 		</div>
+		<div>
+			<label for="event-attendance-mode"><?php esc_html_e( 'Attendance Mode', 'gp-translation-events' ); ?></label>
+				<input type="radio" id="event-attendance-mode-hybrid" name="event_attendance_mode" <?php echo $event->is_hybrid() ? esc_attr( 'checked' ) : ''; ?> value="hybrid" required>
+				<span title="Attendees can attend remotely and on-site"><?php esc_html_e( 'Hybrid (Remote and On-site)', 'gp-translation-events' ); ?></span>
```
Comment (akirk) [inline templates/parts/event-form.php]: If you use a `label` HTML tag like this, the text is clickable to activate the radio box. Please also do this for the other radios.
```suggestion
				<label title="Attendees can attend remotely and on-site"><input type="radio" id="event-attendance-mode-hybrid" name="event_attendance_mode" <?php echo $event->is_hybrid() ? esc_attr( 'checked' ) : ''; ?> value="hybrid" required>
				<?php esc_html_e( 'Hybrid (Remote and On-site)', 'gp-translation-events' ); ?></label>
```

File: templates/event-details.php [hunk 2]
```diff
@@ -288,8 +288,15 @@ function ( $contributor ) {
 					<?php if ( $user_is_attending ) : ?>
 						<input type="submit" class="button is-secondary attending-btn" value="<?php esc_attr_e( "You're attending", 'gp-translation-events' ); ?>" />
 					<?php else : ?>
-						<input type="submit" class="button is-primary attend-btn" value="<?php esc_attr_e( 'Attend Event Onsite', 'gp-translation-events' ); ?>"/>
-						<input type="submit" name="attend_remotely" class="button is-primary attend-btn" value="<?php esc_attr_e( 'Attend Event Remotely', 'gp-translation-events' ); ?>"/>
+						<?php if ( ! $event->is_remote() ) : ?>
+							<input type="submit" class="button is-primary attend-btn" value="<?php esc_attr_e( 'Attend Event On-site', 'gp-translation-events' ); ?>"/>
+							<p class="onsite-btn-note">
```
Comment (akirk) [inline templates/event-details.php]: This label is also shown for Hybrid events but shouldn't be.

**Review by akirk — CHANGES_REQUESTED** [review]
Almost works well, just the message about "only onsite" is displaye for hybrid events also.

File: templates/event-details.php (hunk 2 above)
Comment (trymebytes) [reply templates/event-details.php]: Thanks for catching this

**Review by akirk — APPROVED** [review]
Looks good now!

**Comment by trymebytes** [comment]
Thanks!

**Review by psrpinto — COMMENTED** [review]
Code looks great! 👌

**Pull request merged**